nebula_ros/src/hesai/hw_monitor_wrapper.cpp

```diff
@@ -217,71 +222,53 @@ void HesaiHwMonitorWrapper::OnHesaiLidarMonitorTimer()
   }
 }
 
 void HesaiHwMonitorWrapper::HesaiCheckPtp(diagnostic_updater::DiagnosticStatusWrapper & diagnostics)
 {
+  uint8_t level = diagnostic_msgs::msg::DiagnosticStatus::OK;
+  std::vector<std::string> msg;
   std::scoped_lock lock(mtx_lidar_status_);
   if (current_status_) {
-    uint8_t level = diagnostic_msgs::msg::DiagnosticStatus::OK;
-    std::vector<std::string> msg;
-    auto gps_status = current_status_->get_str_gps_pps_lock();
-    auto gprmc_status = current_status_->get_str_gps_gprmc_status();
-    auto ptp_status = current_status_->get_str_ptp_clock_status();
-    std::transform(gps_status.cbegin(), gps_status.cend(), gps_status.begin(), toupper);
-    std::transform(gprmc_status.cbegin(), gprmc_status.cend(), gprmc_status.begin(), toupper);
-    std::transform(ptp_status.cbegin(), ptp_status.cend(), ptp_status.begin(), toupper);
-    diagnostics.add("gps_pps_lock", gps_status);
-    diagnostics.add("gps_gprmc_status", gprmc_status);
-    diagnostics.add("ptp_clock_status", ptp_status);
-    if (gps_status != "UNKNOWN") {
-      msg.emplace_back("gps_pps_lock: " + gps_status);
-    }
-    if (gprmc_status != "UNKNOWN") {
-      msg.emplace_back("gprmc_status: " + gprmc_status);
-    }
-    if (ptp_status != "UNKNOWN") {
-      msg.emplace_back("ptp_status: " + ptp_status);
-    }
-    if (ptp_status == "FREE RUN" && gps_status == "UNKNOWN") {
-      level = diagnostic_msgs::msg::DiagnosticStatus::WARN;
-    }
-    diagnostics.summary(level, boost::algorithm::join(msg, ", "));
-  } else {
+  json data = current_status_->to_json();
+    for (auto & [key, value] : data.items()) {
+      diagnostics.add(key, value);
+      std::cout << key << " : " << value << std::endl;
+    } 
+  diagnostics.summary(level, boost::algorithm::join(msg, ", "));
+  }else {
     diagnostics.summary(diagnostic_msgs::msg::DiagnosticStatus::WARN, "No data available");
   }
 }
 
-void HesaiHwMonitorWrapper::HesaiCheckTemperature(
-  diagnostic_updater::DiagnosticStatusWrapper & diagnostics)
+void HesaiHwMonitorWrapper::HesaiCheckTemperature(diagnostic_updater::DiagnosticStatusWrapper & diagnostics)
 {
+  uint8_t level = diagnostic_msgs::msg::DiagnosticStatus::OK;
+  std::vector<std::string> msg;
   std::scoped_lock lock(mtx_lidar_status_);
   if (current_status_) {
-    uint8_t level = diagnostic_msgs::msg::DiagnosticStatus::OK;
-    std::vector<std::string> msg;
-    for (size_t i = 0; i < std::size(current_status_->temperature); i++) {
-      diagnostics.add(
-        temperature_names_[i],
-        GetFixedPrecisionString(current_status_->temperature[i].value() * 0.01, 3));
+    json data = current_status_->to_json();
+    for (auto & [key, value] : data.items()) {
+      // Skip execpt temperature
+      if (key != "temperature") {continue;}
+      for (auto & [key, value] : data.items()) {
+        diagnostics.add(key, value);
+        std::cout << key << " : " << value << std::endl;    
+      }
     }
```

Comment:
This nested for-loop is incorrect, it should be a normal for-loop.

For `n` temperature values, this will add **all** diagnostics `n` times, not just the one temperature.